scipy/optimize/_qap.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+from . import linear_sum_assignment, OptimizeResult
+
+
+def umeyama_eigendecomposition(A, B, maximize=False):
+    """Finds an approximate solution to QAP using Umeyama's method."""
+    # 'Correct' non-hermitian matrices
+    if (A != A.T).any() or (B != B.T).any():
+        A = (A + A.T) / 2 + 1j * (A - A.T) / 2
+        B = (B + B.T) / 2 + 1j * (B - B.T) / 2
+
+    va = np.linalg.eigh(A)[1]
+    vb = np.linalg.eigh(B)[1]
+    W = np.abs(vb) @ np.abs(va).T
+    _, col_ind = linear_sum_assignment(W, maximize=maximize)
```

Comment:
After making the corrections above, `W` is correct. But for [2] (original paper) Section IV B, `linear_sum_assignment` is not giving the output reported in the paper.

Reply:
OK. I'll change this and add comments in the code/tests to explain the departure from the convention in the paper.

Reply:
I have changed this now.